x/ecocredit/basket/types/v1/msg_update_curator.go

```diff
@@ -0,0 +1,49 @@
+package v1
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/basket"
+)
+
+var _ legacytx.LegacyMsg = &MsgUpdateCurator{}
+
+// Route implements LegacyMsg.
+func (m MsgUpdateCurator) Route() string { return sdk.MsgTypeURL(&m) }
+
+// Type implements LegacyMsg.
+func (m MsgUpdateCurator) Type() string { return sdk.MsgTypeURL(&m) }
+
+// GetSignBytes implements LegacyMsg.
+func (m MsgUpdateCurator) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&m))
+}
+
+// ValidateBasic does a stateless sanity check on the provided data.
+func (m MsgUpdateCurator) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Curator); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("curator: %s", err)
+	}
+
+	if _, err := sdk.AccAddressFromBech32(m.NewCurator); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf("new curator: %s", err)
+	}
+
+	if m.NewCurator == m.Curator {
+		return sdkerrors.ErrInvalidAddress.Wrap("curator and new curator cannot be the same")
+	}
+
+	if err := basket.ValidateBasketDenom(m.Denom); err != nil {
+		return err
```

Comment:
Might be nice to wrap the parse error in an invalid request error for consistency. Not sure if this is how we want to do things in the long run but we should be consistent for now. See https://github.com/regen-network/regen-ledger/issues/1147#issuecomment-1199793577 for reference.